2011-10-19 Henry Yu <address@hidden>
* NEWS: "rm -FOO" now doesn't suggests "rm ./-FOO" if the file "-FOO"
        exists and "-FOO" is not a valid option.
        * src/rm.c: Include doesn't lstat.h, quotearg.h.
        (diagnose_leading_hyphen): This function is now gone.
        (main): Doesn't Use it.
diff --git a/coreutils-8.0/src/rm.c b/coreutils-8.0/src/rm.c
index 2b62e5b..f1e3540 100644
--- a/coreutils-8.0/src/rm.c
+++ b/coreutils-8.0/src/rm.c
@@ -28,7 +28,6 @@
 #include "argmatch.h"
 #include "error.h"
 #include "quote.h"
-#include "quotearg.h"
 #include "remove.h"
 #include "root-dev-ino.h"
 #include "yesno.h"
@@ -98,33 +97,6 @@ static enum interactive_type const interactive_types[] =
 };
 ARGMATCH_VERIFY (interactive_args, interactive_types);
 
-/* Advise the user about invalid usages like "rm -foo" if the file
-   "-foo" exists, assuming ARGC and ARGV are as with `main'.  */
-
-static void
-diagnose_leading_hyphen (int argc, char **argv)
-{
-  /* OPTIND is unreliable, so iterate through the arguments looking
-     for a file name that looks like an option.  */
-  int i;
-
-  for (i = 1; i < argc; i++)
-    {
-      char const *arg = argv[i];
-      struct stat st;
-
-      if (arg[0] == '-' && arg[1] && lstat (arg, &st) == 0)
-        {
-          fprintf (stderr,
-                   _("Try `%s ./%s' to remove the file %s.\n"),
-                   argv[0],
-                   quotearg_n_style (1, shell_quoting_style, arg),
-                   quote (arg));
-          break;
-        }
-    }
-}
-
 void
 usage (int status)
 {
@@ -309,7 +281,6 @@ main (int argc, char **argv)
         case_GETOPT_HELP_CHAR;
         case_GETOPT_VERSION_CHAR (PROGRAM_NAME, AUTHORS);
         default:
-          diagnose_leading_hyphen (argc, argv);
           usage (EXIT_FAILURE);
         }
     }
